### PR TITLE
Add methods to resume Uploads

### PR DESCRIPTION
**Pull request by tillz**
This Pull Request adds Support to resume an upload to google Drive, as asked in #658.
To achieve this, the following changes were Made:
* Renamed the `getResumeUri()`-Method to `fetchResumeUri()`, as it's more clear
* Added the Method `getResumeUri()` which is now used as a getter for the private `resumeUri`-property. 
* Added a `resume($uri)`-method to be called with an resumeUri, which will fetch the upload status & setup the object (set the resumeUri & progress properties).
* Moved code from `nextChunk()` to the new private `makePutRequest($request)` method to avoid duplicating code from `nextChunk()` in `resume()`.

With this changes, any chunked upload can be easily paused (or a better error-handling can be implemented) by simply getting the resumeURI from the `Google_Http_MediaFileUpload`-Object, storing it anywhere & later creating a new `Google_Http_MediaFileUpload`-Object with the same constructor-parameters as the original one & calling `$object->resume( $previously_stored_resumeUri);`, then proceeding with `nextChunk`. If `nextChunk` is called with chunked data, `$object->getProgress` can be used to find out the offset to be applied to the supplied data.

All informations are from [this reference](https://developers.google.com/drive/web/manage-uploads#resume-upload).

### CODE REVIEW

File: src/Google/Http/MediaFileUpload.php [hunk 1]
```diff
@@ -263,7 +292,12 @@ public function getUploadType($meta)
     return self::UPLOAD_MULTIPART_TYPE;
   }
 
-  private function getResumeUri()
+  public function getResumeUri()
+  {
+    return ( $this->resumeUri !== null ? $this->resumeUri : $this->fetchResumeUri() );
```
Comment (bshaffer) [inline src/Google/Http/MediaFileUpload.php]: This could better be written:

```php
return $this->resumeUri ?: $this->fetchResumeUri();
```

File: src/Google/Http/MediaFileUpload.php [hunk 2]
```diff
@@ -165,7 +146,7 @@ public function nextChunk($chunk = false)
     $response->setExpectedClass($this->request->getExpectedClass());
     $code = $response->getResponseHttpCode();
     $this->httpResultCode = $code;
-
+
```
Comment (bshaffer) [inline src/Google/Http/MediaFileUpload.php]: we need to get rid of unnecessary whitespace

File: src/Google/Http/MediaFileUpload.php (hunk 1 above)
Comment (tillz) [inline src/Google/Http/MediaFileUpload.php]: According to the README.md, PHP 5.2 is supported, but according to [the docs](http://php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary), this was introduced in 5.3?

Comment (bshaffer) [inline src/Google/Http/MediaFileUpload.php]: You are right... we are still supporting 5.2 for this branch. Good catch.

File: src/Google/Http/MediaFileUpload.php [hunk 3]
```diff
@@ -124,37 +124,18 @@ public function getHttpResultCode()
   {
     return $this->httpResultCode;
   }
-
+
```
Comment (bshaffer) [inline src/Google/Http/MediaFileUpload.php]: Please trim this whitespace as well

**Comment by bshaffer** [comment]
Thank you for this PR! Please trim the whitespace and rebase to a single commit, and we should be good to go here.

**Comment by tillz** [comment]
Sorry, i'm not that familiar with rebasing. Can i just close this PR and send a new one, consisting of one commit?
Sidenote: the newlines on blank lines seem to be inconsistent in this repo. I think the reason of this is that the phpcs ruleset accepts whitespaces in blank lines. Possibly setting 'ignoreBlankLines' to false [here](https://github.com/google/google-api-php-client/blob/master/style/ruleset.xml#L52) would solve the problem (and save much work!)

**Comment by bshaffer** [comment]
@tillz you can close this PR. You should never submit PRs from your master branch anyway, but instead should create a feature branch and work off that.

The following should accomplish this:
```
$ git remote add google git@github.com:google/google-api-php-client.git
$ git checkout -b patch-1 google/master
$ git cherry-pick f09b3d7
$ git push origin patch-1
```

Then you just need to submit a new PR from `tillz/patch-1` to `google/master`
